src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -156,6 +156,23 @@ public async Task<IActionResult> ExecuteAsync(FindRequestContext context, string
                                        FormatFindResult(queryJson, context);
         }
 
+        /// <summary>
+        /// Given the FindRequestContext, obtains the query text, executes it against the backend and returns the result as a JsonDocument.
+        /// This method is used for REST POST, PATCH and PUT API requests.
+        /// </summary>
+        public async Task<JsonDocument?> ExecuteAsyncAndGetResponseAsJsonDocument(FindRequestContext context, string datasourceName = "")
```

Comment:
```suggestion
        public async Task<JsonDocument?> ExecuteFollowUpReadAsync(FindRequestContext context, string datasourceName = "")
```

`asJsonDocument` is redundant